.github/workflows/publish-pypi.yml

```diff
@@ -54,7 +54,7 @@ jobs:
   publish:
     name: >-
       Publish Python 🐍 distribution 📦 to PyPI
-    if: github.repository_owner == 'pyopensci'
+    if: github.repository_owner == 'pyopensci' && github.event_name == 'release'
```

Comment:
I noticed that this build step was running for every commit on `main`. We actually only want this running on release events